pkg/backends/newrelic/newrelic.go

```diff
@@ -189,7 +191,29 @@ func (n *Client) processMetrics(metrics *gostatsd.MetricMap, cb func(*timeSeries
 
 // SendEvent sends an event to New Relic.
 func (n *Client) SendEvent(ctx context.Context, e *gostatsd.Event) error {
-	return nil
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case buffer := <-n.eventsBufferSem:
+		defer func() {
+			buffer.Reset()
+			n.eventsBufferSem <- buffer
+		}()
+
+		m := make(map[string][]map[string]string)
```

Comment:
This is a pretty rough type to reason about, and it's leaning more towards the pythonic style.  I would much prefer to see proper structure definitions.